packages/components/pie-modal/src/index.ts

```diff
@@ -24,7 +29,49 @@ export class PieModal extends RtlMixin(LitElement) {
         headingLevel: ModalProps['headingLevel'] = 'h2';
 
     @query('dialog')
-        _dialog: HTMLDialogElement;
+        _dialog!: HTMLDialogElement;
+
+    firstUpdated (changedProperties: DependentMap<ModalProps>) {
+        this._handleModalOpenOnFirstRender(changedProperties);
+    }
+
+    updated (changedProperties: DependentMap<ModalProps>) {
+        this._handleModalStateChanged(changedProperties);
+    }
+
+    connectedCallback () {
+        super.connectedCallback();
+        document.addEventListener(ON_MODAL_OPEN_EVENT, this._disableScrolling);
+        document.addEventListener(ON_MODAL_CLOSE_EVENT, this._enableScrolling);
+    }
+
+    disconnectedCallback () {
+        document.removeEventListener(ON_MODAL_OPEN_EVENT, this._disableScrolling);
+        document.removeEventListener(ON_MODAL_CLOSE_EVENT, this._enableScrolling);
+        this._enableScrolling();
+        super.disconnectedCallback();
+    }
+
+    private _handleModalOpenOnFirstRender (changedProperties: DependentMap<ModalProps>) {
+        // This ensures if the modal is open on first render, the scroll lock is applied
+        if (changedProperties.has('isOpen')) {
+            const previousValue = changedProperties.get('isOpen') as boolean;
+            if (previousValue === undefined && this.isOpen) {
+                this._dispatchModalOpenEvent();
+            }
+        }
+    }
+
+    private _handleModalStateChanged (changedProperties: DependentMap<ModalProps>) {
+        if (changedProperties.has('isOpen')) {
+            const previousValue = changedProperties.get('isOpen') as boolean;
+            if (previousValue) {
+                this._dispatchModalCloseEvent();
+            } else if (previousValue === false) {
```

Comment:
Do you need to specifically check false here? Should be able to just remove the `if` and leave in the else.

Reply:
it can also be `undefined`, in which case we don't want to dispatch anything

Reply:
How does `as boolean` behave when it could be `undefined`?
It could make sense to do an early `return` if it's `undefined`

Reply:
I can't imagine `as boolean` doing us any favours, but an early return sounds like a much better way to do this. Far more explicit!

Reply:
@xander-marjoram even though its cast it seems like it can be undefined 😢

Reply:
<img width="682" alt="Screenshot 2023-06-27 at 11 19 47" src="https://github.com/justeattakeaway/pie/assets/16766185/1e969a09-ea5a-4e89-bca8-72408ab22761">

Reply:
<img width="377" alt="Screenshot 2023-06-27 at 11 22 19" src="https://github.com/justeattakeaway/pie/assets/16766185/325ebaa5-78ee-479c-874e-4ac78e3f3600">
All initial `changedProperties` values on first render

Reply:
Sorry further question because i think we may not even need the `as Boolean` type assertion. The `changedProperties.has()` already has an assertion that a boolean (primitive: true, false) will be returned so I think you just need to do a check against that without asserting again on the `has` method, effectively twice. That way i think it removes the `boolean | undefined` and you wont be left with making further checks later in the code.

You could just stick with the below then in the if statement continue as normal without checking anything else.
` const isClosed = changedProperties.get('isOpen') === false;`